build-logic/publishing/src/main/kotlin/gradlebuild.publish-public-libraries.gradle.kts

```diff
@@ -50,6 +53,35 @@ publishing {
     configurePublishingTasks()
 }
 
+val pgpSigningKey: Provider<String> = providers.environmentVariable("PGP_SIGNING_KEY").forUseAtConfigurationTime()
+val signArtifacts: Boolean = !pgpSigningKey.orNull.isNullOrEmpty()
+
+tasks.withType<Sign>().configureEach { isEnabled = signArtifacts }
+
+signing {
+    useInMemoryPgpKeys(
+        project.providers.environmentVariable("PGP_SIGNING_KEY").forUseAtConfigurationTime().orNull,
+        project.providers.environmentVariable("PGP_SIGNING_KEY_PASSPHRASE").forUseAtConfigurationTime().orNull
+    )
+    publishing.publications.configureEach {
+        if (signArtifacts) {
+            signing.sign(this)
+        }
+    }
+}
+
+fun configureJavadocVariant() {
+    java {
+        withJavadocJar()
+    }
+    tasks.named<Javadoc>("javadoc") {
+        (options as StandardJavadocDocletOptions).apply {
+            addBooleanOption("html4", true)
+            addBooleanOption("Xdoclint:none", true)
```

Comment:
What is the current state for this? Do the published libraries not pass the Javadoc lint checks?